src/coreclr/jit/emitarm64.cpp

```diff
@@ -19048,6 +19075,19 @@ void emitter::emitIns_Call(EmitCallType          callType,
     return (bits << 11); // bits at locations [14,13,12,11]
 }
 
+/*****************************************************************************
+ *
+ *  Returns the encoding for an immediate in the SVE variant of dup (indexed)
+ */
+/*static*/ emitter::code_t emitter::insEncodeSveBroadcastIndex(emitAttr elemsize, ssize_t index)
+{
+    code_t   bits       = (code_t)index;
+    unsigned lane_bytes = genLog2(elemsize) + 1;
+    bits <<= lane_bytes;
+    bits |= (1 << (lane_bytes - 1));
+    return insEncodeSplitUimm<23, 22, 20, 16>(bits);
```

Comment:
i am wondering if we should add an assert inside `insEncodeSplitUimm` to make sure that bits between `lo1` and `hi2` are 0?

Reply:
Yes I think more assertions on these generic functions are usually worth it.